### DISCUSSION TITLE
Resend all failed to certain recipient

### DISCUSSION THREAD

**Discussion by UEAB-BL** [postalserver/postal#3230] [Help with using Postal] [answered]
Hello!
I need to resend all failed, filtered on a specific recipient, and for future use maybe even dates. How do I do this?

The issue comes from a legitimate sending to a certain recipient that unfortunately has an hourly limit per sender and the rest got held. I then released the held using:
`server.message_db.messages(:where => {:held => 1, :scope => 'outgoing'}).each { |m| m.add_to_message_queue(:manual => true) }`
Unfortuneatly that meant that when the limit was reached again, the rest got stuck as failed and I dont know how to resend them. Need some help here rather urgent!

**Comment by UEAB-BL** [comment]
Also, it would really useful with some documentation on the console in general, how do you guys get these commands?

**Reply by willpower232** [reply]
Its all ruby, using the code built into Postal. If you or someone knows ruby then I presume it makes more sense than it does to me :sweat_smile:

**Comment by willpower232** [comment]
Presumably you could update the `:where` 's to include the messages but you'd probably have to have a look in the database to figure out the conditions.

**Reply by UEAB-BL** [reply]
Unfortunetly Im not too familiar with ruby, and neither is anyone I know.
But, I have this so far:
```
Postal console
server.message_db.messages(:where => {:status => 'HardFail', :scope => 'outgoing', :rcpt_to => 'user@domain.com'}).each do |message|
puts message
end
```
That lists the messages I need redelivered, but Im not sure how to mark them for redelivery.
I tried:
```
Postal console
server.message_db.messages(:where => {:status => 'HardFail', :scope => 'outgoing', :rcpt_to => 'user@domain.com'}).each do |message|
message.redeliver
end
```
But it doesnt seem to be working. 
**Do you or anyone else know the correct command to redeliver failed messages?**

I tried looking through some source code but cant find it, dont really know where to look either.

**Reply by UEAB-BL** [reply, marked as answer]
OK, sorry, never mind, I must have done a typo on my first go-around, this ended up working:
```
Postal console
server.message_db.messages(:where => {:status => 'HardFail', :scope => 'outgoing', :rcpt_to => 'user@domain.com'}).each do |message|
message.add_to_message_queue(:manual => true)
end

```
Thanks for the pointers, it got me there! =)
